src/maybe.ts

```diff
@@ -1722,6 +1722,61 @@ export function get<T, K extends keyof T>(
   return curry1(Maybe.andThen(property<T, K>(key)), maybeObj);
 }
 
+/**
+  Transform a function from a normal JS function which may return `null` or
+  `undefined` to a function which returns a `Maybe` instead.
+
+  For example, you might want to wrap the `Document#querySelector` DOM API so
+  you don't have to wrap its results in a `Maybe` every time you use it. Without
+  `wrapReturn`, you always have to explicitly wrap `Document#querySelector`
+  invocations with `Maybe.of`:
+
+  ```ts
+  const aWidth = Maybe.of(document.querySelector('#foo'))
+    .map(el => el.getBoundingClientRect().width)
+    .unwrapOr(0);
+
+  const aColor = Maybe.of(document.querySelector('.bar'))
+    .andThen(
+      el => el instanceof HTMLElement
+        ? Maybe.of(el.style.color)
+        : Maybe.nothing()
+    );
+  ```
+
+  With `wrapReturn`, you can create a transformed version of the function *once*
+  and then be able to use it freely throughout your codebase, *always* getting
+  back a `Maybe`:
+
+  ```ts
+  const querySelector = Maybe.wrapReturn(document.querySelector.bind(document));
```

Comment:
So this is a great example, but I think it gets _much_ more motivating when you have multiple nulls on the way to your destination, so I might change this example to this, or something similar:
```ts
const querySelector = Maybe.wrapReturn(document.querySelector.bind(document));
const getStyle = Maybe.wrapReturn((el: HTMLElement, rule: string) => el.style[rule]);

const aWidth = querySelector('#foo')
  .map(el => el.getBoundingClientRect().width)
  .unwrapOr(0);
	
const aColor = querySelector('.bar')
  .andThen(el => getStyle(el, 'color'))
  .unwrapOr(DEFAULT_COLOR);
```

It would be even better to show how to do this without any maybes at all:
```ts
const bar = document.querySelector('.bar');

if (bar != null) {
  const color = bar.style.color;
  
  if (!color) {
    return DEFAULT_COLOR;
  }
}
```

I think once people start seeing the nesting, that's when it clicks

Reply:
Great suggestion! I'll update it!